core/debugger/remote_debugger_peer.cpp

```diff
@@ -96,11 +98,13 @@ void RemoteDebuggerPeerTCP::_write_out() {
 	while (tcp_client->get_status() == StreamPeerTCP::STATUS_CONNECTED && tcp_client->wait(NetSocket::POLL_TYPE_OUT) == OK) {
 		uint8_t *buf = out_buf.ptrw();
 		if (out_left <= 0) {
-			if (out_queue.is_empty()) {
-				break; // Nothing left to send
-			}
 			mutex.lock();
-			Variant var = out_queue.front()->get();
+			List<Array>::Element *E = out_queue.front();
+			if (!E) {
+				mutex.unlock();
+				break;
+			}
+			Variant var = E->get();
```

Comment:
This was a data-race, as it was reading the `is_empty` outside the mutex. Simplify the code to just pop the front element and check for `nullptr`